packages/docusaurus-theme-classic/src/theme/TOCInline/index.tsx

```diff
@@ -10,39 +10,92 @@ import clsx from 'clsx';
 import type {TOCInlineProps} from '@theme/TOCInline';
 import styles from './styles.module.css';
 import {TOCItem} from '@docusaurus/types';
+import {useThemeConfig} from '@docusaurus/theme-common';
 
 /* eslint-disable jsx-a11y/control-has-associated-label */
 function HeadingsInline({
   toc,
   isChild,
+  isInnerList,
+  maxHeadingLevel,
+  minHeadingLevel,
 }: {
   toc: readonly TOCItem[];
   isChild?: boolean;
+  isInnerList?: boolean;
+  maxHeadingLevel: number;
+  minHeadingLevel: number;
 }) {
-  if (!toc.length) {
+  // if no headings or every heading is too deep, return nothing
+  if (!toc.length || toc.every((heading) => heading.level > maxHeadingLevel)) {
     return null;
   }
-  return (
-    <ul className={isChild ? '' : 'table-of-contents'}>
-      {toc.map((heading) => (
+
+  const prunedTOC = toc.map((heading) => {
```

Comment:
this duplicates algorithmic logic with TOC, we should find a way to factorize it if it can't be simplified

Reply:
I was thinking that should go in a follow-up PR because that duplication happens only once. 😅